.circleci/config.yml

```diff
@@ -5,116 +5,125 @@ jobs:
-      - run: sudo apt-get install cf-cli
-      - run: sudo apt-get install -qy apache2-utils
-      - run: htpasswd -cb Staticfile.auth uikit uikit # the password and username is not a secret. It is just to prevent spiders from indexing
-      - run: cf install-plugin https://github.com/govau/autopilot/releases/download/0.0.5-venapp/autopilot-linux -f
-      - run: cf version                             # log what version we are running
-      - run: cf login -a api.system.y.cld.gov.au -o $CF_ORG_STAGING -s $CF_SPACE_STAGING -u $CF_USER_STAGING -p $CF_PASSWORD_STAGING
-      - run: tree -C                                # log all files before we push
-      - run: cf zero-downtime-push uikit -f manifest-staging.yml
-
+      - checkout                                          # get the files from the repo (why would you ever not want the files????)
+      - run: 
+          name: Writing cache checksum for uikit modules to file, using package-lock.json
+          command: |
+            find packages/*/package-lock.json -type f | md5sum | awk {'print $1'} > /tmp/_PACKAGE_CHECKSUM_FILE
+      - restore_cache:
+          name: Restore package cache
+          keys:
+            - m4-packages-{{ checksum "/tmp/_PACKAGE_CHECKSUM_FILE" }}
+      - run: 
+          name: Install dependencies
+          command: npm install                            # duh
+      - run: 
+          name: Bootstrap the packages
+          command: npm run bootstrap                      # installing all dependencies of all packages
+      - save_cache:
+          name: Save package cache
+          key: m4-packages-{{ checksum "/tmp/_PACKAGE_CHECKSUM_FILE" }}
+          paths:                                          # save_cache doesn't support 'paths' globbing as of 07/2018 :(
+            - packages/accordion/node_modules/
+            - packages/animate/node_modules/
+            - packages/body/node_modules/
+            - packages/breadcrumbs/node_modules/
+            - packages/buttons/node_modules/
+            - packages/callout/node_modules/
+            - packages/control-input/node_modules/
+            - packages/core/node_modules/
+            - packages/cta-link/node_modules/
+            - packages/direction-links/node_modules/
+            - packages/footer/node_modules/
+            - packages/grid-12/node_modules/
+            - packages/header/node_modules/
+            - packages/headings/node_modules/
+            - packages/inpage-nav/node_modules/
+            - packages/keyword-list/node_modules/
+            - packages/link-list/node_modules/
+            - packages/page-alerts/node_modules/
+            - packages/progress-indicator/node_modules/
+            - packages/responsive-media/node_modules/
+            - packages/select/node_modules/
+            - packages/side-nav/node_modules/
+            - packages/skip-link/node_modules/
+            - packages/tags/node_modules/
+            - packages/text-inputs/node_modules/
```

Comment:
hard coding all modules will make it hard when new modules are added. I would try to avoid this

Reply:
I completely agree! CircleCI didn't support path globing at the time of making this PR. 

This seems rectified now, I've made some new changes :smile:

Reply:
**[UPDATE:](https://github.com/govau/uikit/compare/6dd95e672170...f3cc3c365072)** CircleCI still isn't supporting path globbing with the `save_cache` key path list, we NEED to hardcode the module paths for now.

Reply:
I would now question whether the fact we gain some speed here makes up for the fact we now have to hard-code each module...

Reply:
This has been split into a seperate PR. I hope that circle ci updates this in the future and we can have super fast builds without the need for specific directories.